### DISCUSSION TITLE
Supply estimated poses and waypoint(s) to the flight controller

### DISCUSSION THREAD

**Discussion by matanjacoby** [microsoft/AirSim#3618] [Support Q&A] [answered]
Hi,
Is there a simple way to supply the flight controller "online" with estimated poses of the drone and some waypoint(s)?
I have a pose estimation module that gets images as input and predicts the relative poses.
I want to move the drone based on that estimation and some waypoints, and don't want to focus on the guidance and lower level control laws.
So I thought of freezing the simulation, take an image, run pose prediction, then continue with a motion command for some dT and repeat. 

I think #3510 also related..

Can you suggest how to do that and particularly the pose-to-drone-command step? 
Many thanks!

**Comment by jonyMarino** [comment, marked as answer]
Hi @matanjacoby! You can use the PX4 controller if it lets you control the drone with that two variables.

**Reply by matanjacoby** [reply]
Thanks very much @jonyMarino!
